File: src/main/java/org/blacksun/utils/WeightList.java
package org.blacksun.utils;

import org.jetbrains.annotations.NotNull;

import java.util.ArrayList;

public final class WeightList {
    private final ArrayList<Integer> weights;
    private final RandomGenerator random;

    public WeightList(@NotNull int... weights) {
        this.weights = new ArrayList<>();
        for (int weight : weights) {
            this.weights.add(weight);
        }
        random = new RandomGenerator(weights.length);
    }

    public int getWeight() {
        return weights.get(random.next());
    }
}

File: src/main/java/org/blacksun/utils/WeightList.kt
package org.blacksun.utils

import java.util.ArrayList

class WeightList(vararg weights: Int) {
    private val weights = ArrayList<Int>()
    private val random: RandomGenerator

    val weight: Int
        get() = weights[random.next()]

    init {
        for (weight in weights) {
            this.weights.add(weight)
        }
        random = RandomGenerator(weights.size)
    }
}
